src/lib/components/CopyCode/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @fileoverview Element that renders copyable code.
+ */
+
+import {BaseElement} from "../BaseElement";
+
+/**
+ * Renders code block that can easily be copied.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class CopyCode extends BaseElement {
+  constructor() {
+    super();
+    this.copyCode = this.copyCode.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.copyButton = document.createElement("button");
+    this.copyButton.className = "web-copy-code__button";
+    this.copyButton.addEventListener("click", this.copyCode);
+
+    const copyContainer = document.createElement("div");
+    copyContainer.className = "web-copy-code__container";
+    copyContainer.append(this.copyButton);
+
+    this.prepend(copyContainer);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    this.copyButton.removeEventListener("click", this.copyCode);
+  }
+
+  copyCode() {
```

Comment:
tiny style nit, would you mind renaming this function to `onCopyCode` or maybe just `onCopy`? I think elsewhere we've used the `on*` prefix to denote event handlers.

Reply:
I have no qualms renaming it, but if you don't mind me asking, is there a reason to make the change? I completely believe that event handlers (or emitted events) should be prefixed by `on`. However if it's an event internal to the component and not bubbled up, does it still count?

Reply:
oh it's just a code style thing. I tend to always prefix handlers with `on*` so if someone is scanning they can quickly grok that it's a handler and not a utility method ([example](https://github.com/GoogleChrome/web.dev/blob/master/src/lib/components/SideNav/index.js#L192)). Just personal preference. But since it's been done a lot already in the existing code, I was thinking it would be good to keep the pattern for consistency sake.